wunderio/next-drupal-starterkit
# NEX-15: switch from jsonapi to graphql

# Dismiss JSONAPI, switch to GraphQL for content data fetching, add graphql-codegen for type generation

## Drupal side changes:
* Add the `graphql` and `graphql_compose` modules
* Add existing entities to the schema
* Add view to display articles with a GraphQL display, used in client and server side listings
* Remove jsonapi extra modules from composer and recipe, remove patches
* update permissions

## Next.js changes:

* *Major refactoring* of data fetching
* add graphql-codegen to generate types out of the drupal GraphQL schema and the queries/fragments defined in the frontend
* add needed queries to:
   * Get menus
   * get static paths
   * get articles
   * get all paragraphs when rendering `page` and `frontpage` nodes
   * get media
* better structure of the components folder
* removal of most of Zod schemas (still there for webform submissions, and env variables)
* refactor of listings (client and server side)
* refactor of statically generated /all-articles page 
* handling of types for queries and components using fragments and inferred types

## To test

To get to a working site, it should be enough to:
1. check out the repository
2. run `./setup.sh`
